src/client.ts

```diff
@@ -14,6 +14,10 @@ import type { ClickHouseSettings } from './settings'
 import type { InputJSON, InputJSONObjectEachRow } from './clickhouse_types'
 
 export interface ClickHouseClientConfigOptions {
+  /** ClickHouse connection string url
+   * ```clickhouse://[user_info@][hosts_and_ports][/dbname][?query_parameters]```
+   */
+  url?: string
```

Comment:
Considering that `url` might conflict with other params (host, dbname, user, password, params, etc.) and it's easy to make a typo in a string-based configuratuion, I'm not sure it should be supported. This logic might be easily implemented in the application 
```js
const url = new URL('clickhouse://clickhouse://john:secret@127.0.0.1:9000')
const client = createClient({ host: url.host, .... });
```